# **MS-***Pserrata***-SN-2023**
A reposatory providing scripts used in the analysis of genomic and phylogenetic data included in the publication titled "....".

## **Phylogenetic dating**
### **ortho_gen.py**
Used in obtaining sets of orthologs in fasta format corresponding to different arthropods' species from a nexus format ortholog alignment file sourced from [de Moya et al 2020](https://academic.oup.com/sysbio/article/70/4/719/5912026). The script requires [Biopython](https://biopython.org/) for its function. 

## **Population demography**
### **bamcall_multi.sh**
Utilizing [bamCaller.py](https://github.com/stschiff/msmc-tools) to iterate over a list of bam files and defined contigs and generate an output of single VCF and a mask file in bed format per sample ID/contig ID combination.

Requires:
- bamCaller.py
- [utils.py](https://github.com/stschiff/msmc-tools)
- [samtools](http://www.htslib.org/download/)
- [bcftools](http://www.htslib.org/download/)

### **hitsep_multi.sh**
Utilize [generate_multihetsep.py](https://github.com/stschiff/msmc-tools) for parsing and processing VCF and bed files generated by bamcall_multi.sh and organized in separate subfolders based on thier contigs ID. Output multihetsep files are used downstream in [msmc2](https://github.com/stschiff/msmc2) based analysis. 

### **msmc2_multi.sh**
Run [msmc2](https://github.com/stschiff/msmc2) using two different settings to assess the population history interference rubustness and variability under set conditions. The script iterate over bootstrap replicates placed in separate folders and run 5 jobs in parallel with 20 threads set to each job. Results in the output are generated in two separate folders representing each setting used. 

## **Comparative functional genomics**
### **PCoA.R**
R script that utilizes vegan library to calculate Bray-Curtis dissimilarity and perform principal coordinates analysis (PCoA) on the dissimilarity matrix of Pfam and InterProScan hits between compared genomes then vizualize it. 




